src/re_unicode.rs

```diff
@@ -371,6 +371,31 @@ impl Regex {
         Split { finder: self.find_iter(text), last: 0 }
     }
 
+    /// Returns an iterator of substrings of `text` delimited by a match of the
+    /// regular expression. Each element of the iterator will include the
+    /// delimiting match if it appears at the beginning of the element.
+    ///
+    /// This method will *not* copy the text given.
+    ///
+    /// # Example
+    ///
+    /// To split a string delimited by fruit and include the fruit:
+    ///
+    /// ```rust
+    /// # use regex::Regex;
+    /// # fn main() {
+    /// let re = Regex::new(r"(apple|banana|pear)").unwrap();
+    /// let fields: Vec<&str> = re.split("apples: 3 bananas: 2 pears: 4").collect();
+    /// assert_eq!(fields, vec!["apples: 3 ", "bananas: 2 ", "pears: 4"]);
```

Comment:
This does not look like what I'd expect it to be? From the std docs:

> An iterator over substrings of this string slice, separated by characters matched by a pattern. Differs from the iterator produced by split in that split_inclusive leaves the matched part as the terminator of the substring.

The matched part in your expected result is definitely not treated as a terminator here.

And yeah I do kind of think we could do better with an example here. I might copy std's example, but tweak it to match `\r?\n` or something like that.

Reply:
I like the `\r?\n` idea. Gonna make that change....